src/matlab2tikz.m

```diff
@@ -2305,6 +2305,7 @@ function assertRegularAxes(handle)
             userInfo(m2t, '\nMake sure to load \\usepgfplotslibrary{patchplots} in the preamble.\n');
 
             % Default interpolated shader,not supported by polygon, to faceted
+            isFaceColorFlat = isempty(strfind(opts_get(patchOptions, 'shader'),'interp'));
```

Comment:
Well done for spotting. Our current ACID never hits this condition, hence it never raised an error. There is a patch suite that might actually have some plots of this type.